Sources/StreamChat/Controllers/UserListController/UserListController.swift

```diff
@@ -135,13 +147,32 @@ public extension ChatUserListController {
     ///
     func loadNextUsers(
         limit: Int = 25,
-        completion: ((Error?) -> Void)? = nil
+        completion: ((Error?,Bool) -> Void)? = nil
     ) {
+        if hasLoadedAllPreviousChannels {
+            completion?(nil,hasLoadedAllPreviousChannels)
+            return
+        }
         var updatedQuery = query
         updatedQuery.pagination = Pagination(pageSize: limit, offset: users.count)
-        worker.update(userListQuery: updatedQuery) { error in
-            self.callback { completion?(error) }
+        worker.update(userListQuery: updatedQuery) { [weak self] result in
+            guard let weakSelf = self else {
+                completion?(nil,false)
+                return
+            }
+            switch result {
+            case let .success(payload):
+                weakSelf.hasLoadedAllPreviousChannels = payload.users.count < limit
+                weakSelf.callback {
+                    completion?(nil,weakSelf.hasLoadedAllPreviousChannels) }
+            case let .failure(error):
+                weakSelf.callback { completion?(error,weakSelf.hasLoadedAllPreviousChannels)
+                }
+            }
         }
+//        worker.update(userListQuery: updatedQuery) { error in
```

Comment:
@jitendra-simform  can we remove commented code?

Reply:
@parth-simformsolutions  , commented code removed  , please check , thanks